demo/src/pages/forms.rs

```diff
@@ -0,0 +1,235 @@
+	crate::{
+		components::forms::{form_field_wrapper::FormFieldWrapper, form_state_debugger::FormStateDebugger},
+		models::user::{Role, User},
+	}, 
+  dioxus::prelude::*, maestro_forms::fields::{
+		form::{Form, InnerComponentProps},
+		select::SelectFormField,
+		text::TextFormInput,
+		textarea::TextArea,
+	}, 
+  maestro_toast::{
+    ctx::use_toast, toast_info::ToastInfo
+  }, maestro_ui::button::{Button, ButtonSize, ButtonType, ButtonVariant}, strum::VariantNames, tailwind_fuse::tw_join
+};
+
+#[component]
+pub fn FormContent(props: InnerComponentProps<User>) -> Element {
+  let loading = use_signal(|| false);
+  
+  let role_values = Role::VARIANTS.iter().map(|&s| s.to_string()).collect::<Vec<_>>();
+  let role_labels = Role::VARIANTS.iter().map(|&s| s.to_string()).collect::<Vec<_>>();
+
+  let input_class = tw_join!(
+    "w-full p-2 rounded-md border border-gray-300 shadow-sm",
+    "focus:ring-2 focus:ring-blue-400 focus:outline-none placeholder-opacity-50"
+  );
+  
+  rsx! {
+    div { 
+      class: "space-y-6",
+      FormFieldWrapper {
+        label: "Username",
+        field: props.form.get_form_field("username".to_string()),
+        show_validation: true,
+        TextFormInput::<User> {
+          name: "username",
+          placeholder: "Enter your username",
+          class: "{input_class}",
+        }
+      }
+
+      FormFieldWrapper {
+        label: "Email",
+        field: props.form.get_form_field("email".to_string()),
+        show_validation: true,
+        TextFormInput::<User> {
+          name: "email",
+          placeholder: "Enter your email address",
+          class: "{input_class}",
+        }
+      }
+
+      FormFieldWrapper {
+        label: "Bio",
+        field: props.form.get_form_field("bio".to_string()),
+        show_validation: true,
+        TextArea::<User> {
+          name: "bio",
+          placeholder: "Tell us about yourself...",
+          rows: 4,
+          class: "{input_class}",
+        }
+      }
+
+      FormFieldWrapper {
+        label: "Role",
+        field: props.form.get_form_field("role".to_string()),
+        show_validation: true,
+        SelectFormField::<User, String> {
+          name: "role",
+          values: role_values,
+          labels: Some(role_labels),
+          class: "w-full p-2 rounded-md border border-gray-300 focus:ring focus:ring-blue-300 focus:outline-none",
+        }
+      }
+
+      Button {
+				button_type: ButtonType::Submit,
+				disabled: loading(),
+				prevent_default: false,
+				size: ButtonSize::Default,
+				variant: ButtonVariant::Default,
+				class: tw_join!(
+          "mt-4 py-2 rounded-md text-white font-semibold transition-all duration-200",
+          if loading() {
+              "bg-gray-400 cursor-not-allowed opacity-70"
+          } else {
+              "bg-blue-500 hover:bg-blue-600 transform hover:scale-105"
+          }
+      ),
+				if loading() {
+					"Loading..."
+				} else {
+					"Submit"
+				}
+			}
+
```

Comment:
this loading doesn't update because in the component with `on_submit` function there is a local loading state. Plus there are issues with displaying errors: if I submit invalid form and then fix errors and submit a valid one - errors are still present in UI.